sync-android/src/test/java/com/cloudant/sync/datastore/encryption/AndroidKeyProviderTests.java

```diff
@@ -0,0 +1,149 @@
+package com.cloudant.sync.datastore.encryption;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import cloudant.com.androidtest.AndroidTestUtil;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class AndroidKeyProviderTests {
+
+    @Test
+    public void testCreateProviderWithIdentifier() {
+        AndroidKeyProvider provider = new AndroidKeyProvider(AndroidTestUtil.context,
+                ProviderTestUtil.password, ProviderTestUtil.getUniqueIdentifier());
+
+        EncryptionKey createdKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not created", createdKey);
+
+        EncryptionKey loadedKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not loaded", loadedKey);
+
+        assertTrue("loadedKey did not match createdKey", Arrays.equals(createdKey.getKey(),
+                loadedKey.getKey()));
+
+        provider.getManager().clearKey();
+    }
+
+    @Test
+    public void testCreateProviderWithManager() {
+        KeyStorage storage = new KeyStorage(AndroidTestUtil.context, ProviderTestUtil
+                .getUniqueIdentifier());
+        KeyManager manager = new KeyManager(storage);
+        AndroidKeyProvider provider = new AndroidKeyProvider(ProviderTestUtil.password, manager);
+
+        EncryptionKey createdKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not created", createdKey);
+
+        EncryptionKey loadedKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not loaded", loadedKey);
```

Comment:
Would it be possible to check if `provider` is actually calling `manager. loadKeyUsingPassword` the second time `getEncryptionKey` is executed? Maybe passing a mock instead of a proper manager. The test proves that the second time we get the same password but it is not checking what it does.